warp10/src/main/java/io/warp10/script/functions/ASENCODER.java

```diff
@@ -0,0 +1,89 @@
+//
+//   Copyright 2019  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.functions;
+
+import com.google.common.base.Charsets;
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GTSWrapperHelper;
+import io.warp10.continuum.gts.GeoTimeSerie;
+import io.warp10.continuum.store.thrift.data.GTSWrapper;
+import io.warp10.crypto.OrderPreservingBase64;
+import io.warp10.script.ElementOrListStackFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import org.apache.thrift.TDeserializer;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TCompactProtocol;
+
+import java.io.IOException;
+
+/**
+ * Create a GTSEncoder for each instance of GTS, wrapped GTS or raw wrapped GTS.
+ */
+public class ASENCODER extends ElementOrListStackFunction {
```

Comment:
It can also operate on a single GTS which could make `$gts ASENCODERS` a bit misleading... but less misleading than `$lgts ASENCODER` as it is logical that a single GTS produces a single encoder but a list of GTS could produce a single encoder (as `->ENCODER`) or a list of encoders (as this function).

Agreed, I will make it plural.